include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -702,47 +702,43 @@ __parallel_radix_sort(_ExecutionPolicy&& __exec, _Range&& __in_rng)
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng));
         else if (__n <= 1024)
-            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 2>, __wg_size * 2, 8, __radix_bits,
-                                            __is_ascending>{}(__exec.queue(), __in_rng);
+            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 4>, __wg_size * 2, 8, __radix_bits,
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng));
         else if (__n <= 2048)
-            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 3>, __wg_size * 4, 8, __radix_bits,
-                                            __is_ascending>{}(__exec.queue(), __in_rng);
+            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 5>, __wg_size * 4, 8, __radix_bits,
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng));
         else if (__n <= 4096)
-            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 4>, __wg_size * 4, 16, __radix_bits,
-                                            __is_ascending>{}(__exec.queue(), __in_rng);
+            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 6>, __wg_size * 4, 16, __radix_bits,
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng));
         else if (__n <= 8192)
-            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 5>, __wg_size * 8, 16, __radix_bits,
-                                            __is_ascending>{}(__exec.queue(), __in_rng);
+            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 7>, __wg_size * 8, 16, __radix_bits,
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng));
         else if (__n <= 16384)
-            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 6>, __wg_size * 8, 32, __radix_bits,
-                                            __is_ascending>{}(__exec.queue(), __in_rng);
+            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 8>, __wg_size * 8, 32, __radix_bits,
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng));
         else
         {
             assert(__n <= 32768);
-            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 7>, __wg_size * 8, 64, __radix_bits,
-                                            __is_ascending>{}(__exec.queue(), __in_rng);
+            __event = __subgroup_radix_sort<__i_kernel_name<_RadixSortKernel, 9>, __wg_size * 16, 32, __radix_bits,
+                                            __is_ascending>{}(__exec.queue(), ::std::forward<_Range>(__in_rng), __proj);
```

Comment:
I guess `__proj` is left here by mistake.

Reply:
yes